### PR TITLE
Test objects

### PR DESCRIPTION
**Pull request by Christian-B**
Fixes https://github.com/SpiNNakerManchester/sPyNNaker/issues/893

Includes:
https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/775
https://github.com/SpiNNakerManchester/sPyNNaker/pull/1086
https://github.com/SpiNNakerManchester/SpiNNakerGraphFrontEnd/pull/186

tested by:
https://github.com/SpiNNakerManchester/IntegrationTests/pull/57

Renames uinit_test_objects to pacman_test_objects as clearer for (test) imports outside of pacman

Unified test Application vertexes and or moved them to  pacman_test_objects

Simplified SimpleMachineVertex and create MockMAchineVertex for when an API is needed.

### CODE REVIEW

File: pacman/model/graphs/machine/machine_vertex.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def __init__(self, label=None, constraints=None, app_vertex=None,
         :type label: str or None
         :param iterable(AbstractConstraint) constraints:
             The optional initial constraints of the vertex
+        :tpye constraints: iterable(AbstractConstraint)  or None
```
Comment (rowleya) [inline pacman/model/graphs/machine/machine_vertex.py:35]: Typo on :type

File: pacman_test_objects/placer_test_support.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from pacman.model.graphs.common import Slice
+from pacman.model.graphs.machine import SimpleMachineVertex
+from pacman.model.resources import (
+    ConstantSDRAM, CPUCyclesPerTickResource, DTCMResource, ResourceContainer)
+
+
+def get_resourced_machine_vertext(lo_atom, hi_atom, label=None):
```
Comment (rowleya) [inline pacman_test_objects/placer_test_support.py:22]: Should this be get_resourced_machine_vertex?

**Review by rowleya — APPROVED** [review]
Fine, couple of typos...

File: pacman/model/graphs/machine/machine_vertex.py (hunk 1 above)
Comment (Christian-B) [reply pacman/model/graphs/machine/machine_vertex.py:35]: fixed

File: pacman_test_objects/placer_test_support.py (hunk 2 above)
Comment (Christian-B) [reply pacman_test_objects/placer_test_support.py:22]: fixed

**Pull request merged**